docs/DMS/DOCUMENT-AUTHORIZATION.md

```diff
@@ -0,0 +1,170 @@
+# Document Authorization
+
+## Problem
+
+The ODS/API provides several document security configurations. Among those are EducationOrganization-based
+security and Namespace-based security. These security configurations rely on comparing attributes of a
+document (e.g. SchoolId, EducationServiceCenterId, Namespace) with a list of authorizations configured for
+each client. Additionally, EducationOrganizations form a hierarchy (e.g. Schools are a part of a
+LocalEducationAgency) where a higher-level client authorization applies to lower level security attributes.
+
+The design of Client Authorizations is covered in [DMS-AUTH](DMS-AUTH.md).
+
+The scope of the problem is to determine how the DMS will:
+
+- Efficiently extract the security attributes of a document.
+- Understand the hierarchy of EducationOrganizations.
+- Efficiently apply a client's authorizations to POST, PUT, GET and DELETE actions, including to a search
+  engine datastore.
+
+## Solution
+
+### Security Attribute Location
+
+DMS needs to know which resources have security attributes and where those attributes are located on the
+document. These attributes then need to be extracted into a `securityAttributes` object.
+
+MetaEd will provide this information in the ResourceSchema in ApiSchema.json for any EducationOrganizations
```

Comment:
- Should discuss how DMS will validate relationship to students. Not all student related resources have direct EducationOrganizationId field, DMS should traverse that relationship via student school association. In ODS API StudentUSI is considered as a property that falls under authorization context. 

- In ODS API not all role named EducationOrganizationId fields are used for authorization. Typically only fields named EducationOrganizationId is used for auth. Additional override is applied in specific cases for rolenamed EducationOrganizationId fields (e.g. OrganizationDepartment, DisciplineAction, StudentAssessment). It is also possible that a resource has a EducationOrganizationId and another role named EducationOrganizationId and only EducationOrganizationId  is used for authorization e.g. studentProgramEvaluations has EducationOrganization reference and a programEvaluationEducationOrganization reference. 

- This may be already in radar but was not clear to me in the document, identifying the security attributes is important but implementors do need the flexibility to configure how a resource is authorized. Having a namespace field shouldn't enforce namespace based authorization. e.g. all descriptor reads are allowed without checking for API client's namespace prefix.

Reply:
Updated with EdOrg -> School relationships, thank you for highlighting that. Also mentions that configuration will allow enforcement to vary,